### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: src/Gir/Generation/IGeneratableExtensions.cs [hunk 1]
```diff
@@ -50,6 +50,16 @@ static void GenerateImport (this ICallable callable, IGeneratable parent, Indent
 			var retType = GetReturnCSharpType (callable, writer);
 
 			var (typesAndNames, names) = BuildParameters (callable, writer.Options, appendInstanceParameters: true);
+
+			// TODO: Better than using the constant string, insert a custom generatable which contains the import string as a constant.
+			// Also, glib-2.0.gir is problematic, it contains 2 library names :(
```
Comment (stsundermann) [inline src/Gir/Generation/IGeneratableExtensions.cs]: Usually everything is in the first specified library

Comment (stsundermann) [reply src/Gir/Generation/IGeneratableExtensions.cs]: I think I even talked about that issue with the gi devs and they said the same.

File: src/Gir/Generation/Callback.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+				using (writer.Indent ()) {
+					// TODO: Validation if we have an userdata parameter we can use so we can write this code.
+					// Otherwise, we need to generate code which uses a non-static callback.
+
+					var returnType = GetReturnType (writer);
+					
+					// Public API delegate which uses managed types.
+					writer.WriteLine ("[UnmanagedFunctionPointer (CallingConvention.Cdecl)]");
+					writer.WriteLine ($"public delegate {returnType} {Name} PARAMS");
+					writer.WriteLine ();
```
Comment (decriptor) [inline src/Gir/Generation/Callback.cs]: Why not use https://github.com/mono/gir-sharp/blob/master/src/Gir/Generation/IGeneratableExtensions.cs#L100 ?

File: src/Gir/Generation/IGeneratableExtensions.cs (hunk 1 above)
Comment (Therzok) [reply src/Gir/Generation/IGeneratableExtensions.cs]: Cool, we can do that.

File: src/Gir/Generation/Callback.cs (hunk 2 above)
Comment (Therzok) [reply src/Gir/Generation/Callback.cs]: I wrote this for now, need to hack out the `import` definition out of `ICallable`

Comment (Therzok) [reply src/Gir/Generation/Callback.cs]: And a callback is not really a member, it's a type.

**Comment by Therzok** [comment]
Should be better now.

**Pull request merged**